pandas/core/base.py

```diff
@@ -1035,7 +1035,13 @@ def __iter__(self):
         (for str, int, float) or a pandas scalar
         (for Timestamp/Timedelta/Interval/Period)
         """
-        return iter(self.tolist())
+        # We are explicity making element iterators.
+        if is_datetimelike(self._values):
+            return map(com.maybe_box_datetimelike, self._values)
+        elif is_extension_array_dtype(self._values):
+            return iter(self._values)
+        else:
+            return map(self._values.item, range(self._values.size))
```

Comment:
Ha, I am looking at this now and I do not get anymore why we are mapping over a range here? We should map over values of `self._values` to cast them to Python using `self._values.item`. Was this my original code?

Reply:
Yeah, [here's](https://github.com/pandas-dev/pandas/commit/2773e3473cbf0a6cc875dc7c065172b7656f1f53) your commit.

Reply:
Oh, I know. `item` wants an index of the item you want to get back.